inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -638,7 +638,35 @@ void ReversePermutationsPass::run() {
 
 void RemovePermutationsNHWCToNCHWPass::run() {
     std::list<CNNLayerPtr> permutationsToRemove;
-
+    auto isPermuteToRemove = [](CNNLayerPtr layer, bool before) {
+        if (LayerInfo(layer).isPermute()) {
+            auto in_dims = layer->insData[0].lock()->getDims();
+            auto out_dims = layer->outData[0]->getDims();
+            auto check_permute = [in_dims, out_dims](const std::vector<size_t>& permute) {
```

Comment:
Instead of using in_dims and out_dims use layer->GetParamAsInts("order") and comapre order since (1 1 10 10) might be either 0 1 2 3 or 0 1 3 2